scripts/rpm/post-install.sh

```diff
@@ -48,41 +24,15 @@ if [[ ! -f /etc/telegraf/telegraf.conf ]] && [[ -f /etc/telegraf/telegraf.conf.s
    cp /etc/telegraf/telegraf.conf.sample /etc/telegraf/telegraf.conf
 fi
 
+# Set up log directories
+LOG_DIR=/var/log/telegraf
 test -d $LOG_DIR || mkdir -p $LOG_DIR
 chown -R -L telegraf:telegraf $LOG_DIR
 chmod 755 $LOG_DIR
 
-# Distribution-specific logic
-if [[ -f /etc/redhat-release ]] || [[ -f /etc/SuSE-release ]]; then
-    # RHEL-variant logic
-    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    else
-        # Assuming SysVinit
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
-        else
-            install_chkconfig
-        fi
-    fi
-elif [[ -f /etc/os-release ]]; then
-    source /etc/os-release
-    if [[ "$NAME" = "Amazon Linux" ]]; then
-        # Amazon Linux 2+ logic
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    elif [[ "$NAME" = "Amazon Linux AMI" ]]; then
-        # Amazon Linux logic
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
-        else
-            install_chkconfig
-        fi
-    elif [[ "$NAME" = "Solus" ]]; then
-        # Solus logic
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    fi
+# Set up systemd service
+if [[ -d /run/systemd/system ]]; then
```

Comment:
I double checked that checking for this directory is a reliable way to know if systemd is present. I found https://www.freedesktop.org/software/systemd/man/sd_booted.html which says that's how the systemd sd_booted() function does it.

Could you put a link to that page in a comment here so others won't wonder why this directory shows systemd is running?

Reply:
ah perfect that is the same doc I came across, I've added it as a comment.